app/src/main/java/com/example/campusguide/directions/ChooseOriginOptions.kt

```diff
@@ -80,13 +94,26 @@ class ChooseOriginOptions(
         val search = CustomSearch(act, provider, Constants.ORIGIN_SEARCH_REQUEST_CODE)
 
         search.setLocationListener {searchLocation ->
-            val location = Location(searchLocation.name)
-            location.latitude = searchLocation.lat
-            location.longitude = searchLocation.lon
-            locationSelectedListener(location)
+            if(searchLocation != null) {
+                val location = Location(searchLocation.name)
+                location.latitude = searchLocation.lat
+                location.longitude = searchLocation.lon
+                locationSelectedListener(location)
+            }
             act.removeActivityResultListener(search)
         }
         act.addActivityResultListener(search)
         search.openCustomSearchActivity()
     }
+
+    private fun useLastEventLocation(){
+        val calendarBox: Box<Calendar> = ObjectBox.boxStore.boxFor()
+        val mycal = Pair(calendarBox.all[0].id, calendarBox.all[0].name)
+
+        var lastLocation = Events(activity as MapsActivity, mycal).getLastEventLocation()
+
+        GlobalScope.launch {
+            FindEventLocation(activity as FragmentActivity, locationSelectedListener).getLocationOfEvent(lastLocation)
+        }
+    }
 }
```

Comment:
Why are there no new lines on any of these files...?

Reply:
Might fix later, not important.